test/parallel/test-worker-connection.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const common = require('../common');
+const {
+  connect,
+  setConnectionsListener,
+  workerData,
+  Worker,
+  threadId,
+} = require('node:worker_threads');
+const { strictEqual, deepStrictEqual } = require('node:assert');
+
+// Spawn threads on three levels: 1 main thread, two children, four grand childrens. 7 threads total, max id = 6
+const MAX_THREAD = 6;
+const count = workerData?.count ?? 0;
+
+if (count < 2) {
+  new Worker(__filename, { workerData: { count: count + 1 } });
+  new Worker(__filename, { workerData: { count: count + 1 } });
+}
+
+// Do not use mustCall here as the thread might not receive any connection request
+setConnectionsListener(function(source, port, data) {
+  // Let's verify the source hierarchy
+  strictEqual(data.count, source < 3 ? Math.ceil(source / 2) : 2);
+
+  // Verify communication
+  port.on('message', common.mustCall(function(message) {
+    deepStrictEqual(message, { message: 'ping', source, destination: threadId });
+
+    setTimeout(() => {
+      port.postMessage({ message: 'pong', source: threadId, destination: source });
+    }, common.platformTimeout(threadId + 200));
+  }));
+
+  return true;
+});
```

Comment:
Can you please add a test if this function throws and the thread crashes?

Reply:
I did it now.